src/Rings/FreeAssAlgIdeal.jl

```diff
@@ -37,16 +43,20 @@ Return the two-sided ideal of $R$ generated by $g$.
 """
 function ideal(R::FreeAssAlgebra, g::Vector{<:FreeAssAlgElem})
   @assert all(x -> parent(x) == R, g) "parent mismatch"
-  return FreeAssAlgIdeal(R, g)
+  return FreeAssAlgIdeal(IdealGens(R, g))
 end
 
 function ideal(g::Vector{<:FreeAssAlgElem})
   @assert length(g) > 0 "cannot infer base ring"
   algebra = parent(g[1]) 
   return ideal(algebra, g)
+end
 
+function ideal(g::IdealGens{T}, check_groebner_basis::Bool=false) where T <: FreeAssAlgElem
+  return FreeAssAlgIdeal(g, check_groebner_basis)
```

Comment:
```suggestion
function ideal(g::IdealGens{T}; check_groebner_basis::Bool=false) where T <: FreeAssAlgElem
  return FreeAssAlgIdeal(g; check_groebner_basis)
```
for readability, this should better be a kwarg

Reply:
Somehow I didn't see those comments